test/buy-tshirt.spec.ts

```diff
@@ -1,41 +1,45 @@
-import { $, browser } from 'protractor';
+import { browser } from 'protractor';
+import { MenuContentPage, ProductList, ProductAddedModal,
+  SummaryStep, SignInStep, AddressStep,
+  ShippingStep, OrderSummary, BankPayment, PaymentStep} from '../src/page';
 
 describe('Buy a t-shirt', () => {
   beforeEach(() => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
   });
-
+  const menuContentPage: MenuContentPage = new MenuContentPage();
+  const productList: ProductList = new ProductList();
+  const productAddedModal: ProductAddedModal = new ProductAddedModal();
+  const summaryStep: SummaryStep = new SummaryStep();
+  const signInStep: SignInStep = new SignInStep();
+  const addressStep: AddressStep = new AddressStep();
+  const shippingStep: ShippingStep = new ShippingStep();
+  const orderSummary: OrderSummary = new OrderSummary();
+  const bankPayment: BankPayment = new BankPayment();
+  const paymentStep: PaymentStep = new PaymentStep();
   it('then should be bought a t-shirt', async () => {
     await browser.get('http://automationpractice.com/');
-    await(browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
     await(browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+    await menuContentPage.goToTShirtMenu();
     await(browser.sleep(3000));
-    await $('[style*="display: block;"] .button-container > a').click();
+    await productList.addToCart();
     await(browser.sleep(3000));
-    await $('.cart_navigation span').click();
+    await productAddedModal.clickModalBtn();
     await(browser.sleep(3000));
-
-    await $('#email').sendKeys('aperdomobo@gmail.com');
-    await $('#passwd').sendKeys('WorkshopProtractor');
-    await $('#SubmitLogin > span').click();
+    await summaryStep.clickCheckOut();
     await(browser.sleep(3000));
-
-    await $('#center_column > form > p > button > span').click();
+    await signInStep.performLogin();
     await(browser.sleep(3000));
-
-    await $('#cgv').click();
+    await addressStep.clickCheckOut();
     await(browser.sleep(3000));
-
-    await $('#form > p > button > span').click();
+    await shippingStep.acceptTermsOfService();
     await(browser.sleep(3000));
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+    await shippingStep.clickCheckOut();
     await(browser.sleep(3000));
-    await $('#cart_navigation > button > span').click();
+    await orderSummary.clickPayByBank();
```

Comment:
the orderSummaryPage should be the last step, the step to implement here is the paymentStep. I guess you confused the payment step with the orderSummary step.

Reply:
Yes, I for some reason I mistake them. Sorry